### ISSUE TITLE
Add an optarg to `.changes` to alter the limit of changes cached on server

### ISSUE THREAD

**Issue by larkost** [rethinkdb/rethinkdb#3607]
The server will cache 100,000 changes before it starts forgetting newer changes and sending the client a warning message about this. However, that is a huge number and is impractical for some cases, such as:
1. Testing: it takes many minutes to write that many changes to disk on some systems. So at the moment we are not able to test this every time.
2. Cases where the documents may be large will cause the memory usage on the server to ballon, especially since we keep before and after.
3. In some instances data that old might be worthless, so we are just making the client work harder when it has gotten behind.

So I would like to request an optarg to set the number of values to cache for this query. A compiled-in value has also been suggested, but for testing I eventually want to support running the suite on an already-installed version, making that a sub-optimal solution.

**Comment by larkost** [comment]
Once this is done then the sub-test should be updated and enabled in `test/rql_test/src/changefeeds/table.yaml`

**Comment by danielmewes** [comment]
Incidentally, such a parameter is part of the agreed upon proposal for https://github.com/rethinkdb/rethinkdb/issues/3471 .

Leaving this open as a separate issue because it's a different use case, but we'll likely solve both together.

**Comment by danielmewes** [comment]
This parameter exists as `changefeed_queue_size` (not sure which commit added it).
